nntrainer/tensor/manager.h

```diff
@@ -241,11 +245,14 @@ class Manager {
                  bool shared_var = true, bool shared_grad = true);
 
   /**
-   * @brief     Get all the weights
+   * @brief     Get all the weights which match the above condition
    *
-   * @return    return all the weights
+   * @return    return the weights with satisfying the above condition
    */
-  std::vector<Weight *> getWeights();
+  std::vector<Weight *>
+  getWeights(const std::function<bool(Weight *)> &condition = [](Weight *) {
+    return true;
+  });
```

Comment:
```suggestion
  getWeights(const std::function<bool(Weight *)> &condition = nullptr;
```

Reply:
Return true was for expressiveness but nullptr works too.

Reply:
I think this is just minor performance measure. but we will need to handle nullptr case anyway inside the implemenation :)

Reply:
nullptr support is added :)